kernel/comps/virtio/src/device/input/device.rs

```diff
@@ -167,25 +167,46 @@ impl InputDevice {
         }
     }
 
-    /// Query a specific piece of information by `select` and `subsel`, and write
-    /// result to `out`, return the result size.
-    pub fn query_config_select(&self, select: InputConfigSelect, subsel: u8, out: &mut [u8]) -> u8 {
+    pub fn query_config_id_name(&self) -> String {
+        let size = self.select_config(InputConfigSelect::IdName, 0);
+
+        let out = {
+            // TODO: Add a general API to read this byte-by-byte.
+            let mut out = Vec::with_capacity(size);
+            let mut data_ptr = field_ptr!(&self.config, VirtioInputConfig, data).cast::<u8>();
+            for _ in 0..size {
+                out.push(data_ptr.clone().read_once().unwrap());
```

Comment:
It seems that you use `clone().read_once()` all the time, so I guess the `clone()` is actually necessary. But why is this `clone()` needed?

AFAICK, `read_once()` takes `&self`, so removing the `clone()` seems fine.